test/libsolutil/DominatorTest.cpp

```diff
@@ -0,0 +1,876 @@
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["B"])) == std::vector<TestVertexLabel>({"A"}));
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["C"])) == std::vector<TestVertexLabel>({"B", "A"}));
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["D"])) == std::vector<TestVertexLabel>({"B", "A"}));
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["E"])) == std::vector<TestVertexLabel>({"D", "B", "A"}));
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["F"])) == std::vector<TestVertexLabel>({"B", "A"}));
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["G"])) == std::vector<TestVertexLabel>({"C", "B", "A"}));
+	BOOST_TEST(toVertexLabel(dominatorFinder.dominatorsOf(test.verticesIdMap["H"])) == std::vector<TestVertexLabel>({"G", "C", "B", "A"}));
+}
+
+BOOST_AUTO_TEST_CASE(check_dominance)
+{
+	//            A
+	//            │
+	//            ▼
+	//        ┌───B
+	//        │   │
+	//        ▼   │
+	//        C ──┼───┐
+	//        │   │   │
+	//        ▼   │   ▼
+	//        D◄──┘   G
+	//        │       │
+	//        ▼       ▼
+	//        E       H
+	//        │       │
+	//        └──►F◄──┘
+	DominatorFinderTest test(
+		{"A", "B", "C", "D", "E", "F", "G", "H"},
+		{
+			Edge("A", "B"),
+			Edge("B", "C"),
+			Edge("B", "D"),
+			Edge("C", "D"),
+			Edge("C", "G"),
+			Edge("D", "E"),
+			Edge("E", "F"),
+			Edge("G", "H"),
+			Edge("H", "F")
+		},
+		{},
+		{
+			{"A", 0},
+			{"B", 1},
+			{"C", 2},
+			{"D", 3},
+			{"E", 4},
+			{"F", 5},
+			{"G", 6},
+			{"H", 7}
+		},
+		{}
+	);
+
+	// Helper function to create a dominance relation for a vertex with all other vertices.
+	//
+	// @param _indices: The indices of the vertices that the vertex dominates.
+	// @return: A vector of booleans where the index represents the vertex and the value
+	// represents if the vertex dominates the other vertex at that index.
+	auto makeDominanceVertexRelation = [&](std::vector<size_t> const& _indices = {}){
+		std::vector<bool> dominance(test.vertices.size(), false);
+		for (size_t i: _indices)
+		{
+			soltestAssert(i < test.vertices.size());
+			dominance[i] = true;
+		}
+		return dominance;
+	};
+
+	// Dominance truth table for all vertices.
+	// Note that it includes self-dominance relation.
+	std::vector<std::vector<bool>> expectedDominanceTable = {
+		std::vector<bool>(test.vertices.size(), true),    // A dominates all vertices, including itself
+		makeDominanceVertexRelation({1,2,3,4,5,6,7}), // B, C, D, E, F, G, H
+		makeDominanceVertexRelation({2, 6, 7}), // C, G, H
+		makeDominanceVertexRelation({3, 4}),    // D, E
+		makeDominanceVertexRelation({4}),       // E
+		makeDominanceVertexRelation({5}),       // F
+		makeDominanceVertexRelation({6, 7}),    // G, H
+		makeDominanceVertexRelation({7})        // H
+	};
```

Comment:
After looking through tests again now, I realized that it might have been better to have the dominance table as a part of normal expectations (or even automatically calculate it from these expectations we already have). Then we could test `dominatorsOf()` and  `dominates()` in every test with a single call.

It's not critical to have and I'd merge this PR even without it, but because of this, these two are not getting as much coverage as the rest of the code.